packages/edit-site/src/components/page-templates-template-parts/index.js

```diff
@@ -338,16 +341,31 @@ export default function PageTemplatesTemplateParts( { postType } ) {
 		return filterSortAndPaginate( records, view, fields );
 	}, [ records, view, fields ] );
 
-	const editTemplateAction = useEditPostAction();
+	const onActionPerformed = useCallback(
+		( actionId, items ) => {
+			if ( actionId === 'edit-post' ) {
+				const post = items[ 0 ];
+				history.push( {
+					postId: post.id,
+					postType: post.type,
+					canvas: 'edit',
+				} );
+			}
+		},
+		[ history ]
+	);
+	const postActions = usePostActions( { onActionPerformed } );
 	const actions = useMemo(
 		() => [
-			editTemplateAction,
+			postActions.find( ( action ) => action.id === 'edit-post' ),
```

Comment:
Should `usePostActions` accept an array of action IDs, so that we don't have to do this?

```js
const [ editAction, viewRevisionsAction ] = usePostActions(
  { onActionPerformed },
  [ 'edit-post', 'view-post-revisions' ]
);
const actions = useMemo(
  () => [ editAction, resetTemplateAction, /* ... */ ],
  [ editAction, viewRevisionsAction ]
);
```